### PR TITLE
added classification model and optimized fastanpr modules

### PR DESCRIPTION
**Pull request by bonj4**
- padded 10px with the original image
- added classifier class
- and renamed some variables

### CODE REVIEW

**Review by gorkemgul — APPROVED** [review]
Ready to merge :cat:

File: fastanpr/fastanpr.py [hunk 1]
```diff
@@ -9,14 +9,14 @@
 class FastANPR:
     def __init__(
             self,
-            detection_model: Union[str, Path] = Path(__file__).parent / 'best.pt',
+            detection_model: Union[str, Path] = Path(__file__).parent / 'detection_model.pt',
```
Comment (ozcan-durak) [inline fastanpr/fastanpr.py:12]: where is the file ?

File: fastanpr/recognition.py [hunk 2]
```diff
@@ -16,7 +16,12 @@ class Recogniser:
     def __init__(self, device: str):
         self.device = device
         self.model = PaddleOCR(
-            lang="en", use_angle_cls=True, show_log=False, use_gpu=False if device == "cpu" else device
+            use_angle_cls=False, lang='en',
+            # ocr_version='PP-OCRv4',
+            rec_model_dir="/home/visio-ai/PycharmProjects/anpr_api/inference/en_PP-OCRv4_rec",
```
Comment (ozcan-durak) [inline fastanpr/recognition.py:21]: !

File: fastanpr/classification.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            # 1 equals to entries/front, 2 equals to exits/back
+            class_names.append(1 if classes[highest_prob_index] == 'entries' else 2)
+        return class_names
+
+if __name__ == '__main__':
+    classifier = Classifier(device='cuda:0')
+
+    img=cv2.imread('filename.jpg')
+    names= classifier.run([img])
+    print(names)
```
Comment (ozcan-durak) [inline fastanpr/classification.py:33]: why printing names ?

File: fastanpr/fastanpr.py (hunk 1 above)
Comment (gorkemgul) [reply fastanpr/fastanpr.py:12]: ? @bonj4

File: fastanpr/recognition.py (hunk 2 above)
Comment (gorkemgul) [reply fastanpr/recognition.py:21]: !!! @bonj4

File: fastanpr/classification.py (hunk 3 above)
Comment (bonj4) [reply fastanpr/classification.py:33]: it returns the class name. such as 'entries' or 'exists'

File: fastanpr/recognition.py [hunk 4]
```diff
@@ -16,7 +16,12 @@ class Recogniser:
     def __init__(self, device: str):
         self.device = device
         self.model = PaddleOCR(
-            lang="en", use_angle_cls=True, show_log=False, use_gpu=False if device == "cpu" else device
+            use_angle_cls=False, lang='en',
+            # ocr_version='PP-OCRv4',
+            rec_model_dir="/path/to/inference/en_PP-OCRv4_rec",
```
Comment (ozcan-durak) [inline fastanpr/recognition.py:21]: Does user have to know where the path initialise inside the function ? Args or ENV ?

File: fastanpr/classification.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
+from pathlib import Path
+from ultralytics import YOLO
+
+from typing import Union, List
+import cv2
+import numpy as np
+
+
+class Classifier:
+    def __init__(self, device: str ,classification_model: Union[str, Path] = Path(__file__).parent / 'classification_model.pt'):
```
Comment (ozcan-durak) [inline fastanpr/classification.py:10]: Same story here, Do i have know where the path is if i'd like to use different classification model ?

File: fastanpr/fastanpr.py (hunk 1 above)
Comment (ozcan-durak) [reply fastanpr/fastanpr.py:12]: read the path from args or env

**Pull request closed without merging**